jose-elias-alvarez/buftabline.nvim
# Adds option to use buffer ID as index

Hi and thanks for this cool plug-in : )

**Context**

I have a use case in mind where I like to know the buffer ID number.

This allows users to run commands for the specific buffer. For example, easily load a buffer into a split with `:sb 4` where 4 is the buffer ID number.

**Problem**

Currently _buftabline.nvim_ only shows the sequential index in the tabline…

**Approach**

… so I added a `buffer_id_index` option.

**How to test it**

If this is not clear, let me create an example:

1. Open `nvim`
2. Start a new file/buffer `:e testfile`
3. Start another file/buffer `:e testfile2`
4. Close the last buffer `:bw`
5. And start yet another one `:e testfile3`

With the current and default options, the tabline should look like:

```
1. testfile   2. testfile3
```

With `buffer_id_index = true` it should result in something like:

```
1. testfile   3. testfile3
```

**Doubts**

I am pretty new to Lua and I couldn't make the tests run, but I did my best to add a test case.

<details>

```
$ make test
nvim --headless --noplugin -c "lua require(\"plenary.test_harness\").test_directory_command('test {minimal_init = \"test/minimal.vim\"}')"
Starting...Scheduling: test/bufferline_spec.lua
Scheduling: test/buffers_spec.lua
Scheduling: test/set-hlgroup_spec.lua
Scheduling: test/options_spec.lua
Scheduling: test/auto-hide_spec.lua
Scheduling: test/buftabline_spec.lua
Scheduling: test/set-maps_spec.lua
Scheduling: test/add-commands_spec.lua
Scheduling: test/utils_spec.lua
[nvim-treesitter] [0/1] Downloading...
[nvim-treesitter] [0/1] Creating temporary directory
[nvim-treesitter] [0/1] Extracting...
[nvim-treesitter] [0/1] Compiling...
Error detected while processing command line:
Error during compilation
```

</details>

Is that useful for the plugin? I'm happy to help : )

Just let me know and I'll try to find out why test is not running.
